middle_end/flambda2/types/provers.ml

```diff
@@ -771,19 +776,24 @@ let meet_project_function_slot_simple env ~min_name_mode t function_slot :
   | Naked_nativeint _ | Rec_info _ | Region _ ->
     wrong_kind "Value" t
 
-let meet_project_value_slot_simple env ~min_name_mode t env_var :
-    Simple.t meet_shortcut =
+let meet_project_value_slot_simple env ~min_name_mode ~value_slot_kind t env_var
```

Comment:
Here `env_var` already carries a kind, so the new check is still welcome but I don't think we need the new parameter.